### PR TITLE
Firefox 78

### PR DESCRIPTION
**Pull request by cgrzemba**
This is the update to FF 78.9.0esr. For build is setting in /etc/system set _userlimit=0x7fffc0000000
also neccessary.

### CODE REVIEW

**Comment by AndWac** [comment]
With #6640 has been merged (and installed on the build server) everything should be prepared for this.
Please squash the commits.

**Comment by AndWac** [comment]
Does not build here:
checking for cbindgen... no
ERROR: Cannot find cbindgen. Please run `mach bootstrap`,
`cargo install cbindgen`, ensure that `cbindgen` is on your PATH,
or point at an executable with `CBINDGEN`.

**Comment by cgrzemba** [comment]
For cbindgen we have:

    ${HOME}/.cargo/bin/cbindgen:
	cargo install cbindgen

in build recipe but I see nowhere PATH setting for ~/.cargo/bin. I will check who its works in my environment.

**Comment by cgrzemba** [comment]
I moved the cbindgen install step to COMPONENT_PRE_CONFIGURE_ACTION. This works.

**Comment by AndWac** [comment]
It gets further but is still failing:
/usr/bin/touch /oibuild/oi-userland/components/web/firefox/firefox-78.9.0/.patched-rust_time_sys.rs.patch
/usr/bin/touch /oibuild/oi-userland/components/web/firefox/firefox-78.9.0/.patched
/usr/bin/touch /oibuild/oi-userland/components/web/firefox/firefox-78.9.0/.prep
(/bin/rm -f -rf /oibuild/oi-userland/components/web/firefox/build/amd64 ; /bin/mkdir -p /oibuild/oi-userland/components/web/firefox/build/amd64)
rm -f  /oibuild/oi-userland/components/web/firefox/firefox-78.9.0/mozconfig; echo "ac_add_options MOZILLA_OFFICIAL=1" >> /oibuild/oi-userland/components/web/firefox/firefox-78.9.0/mozconfig ; echo "ac_add_options PKG_CONFIG=/usr/bin/amd64/pkg-config" >> /oibuild/oi-userland/components/web/firefox/firefox-78.9.0/mozconfig ; echo "ac_add_options CC_TYPE=gcc" >> /oibuild/oi-userland/components/web/firefox/firefox-78.9.0/mozconfig ; echo "ac_add_options CC=/usr/gcc/7/bin/gcc" >> /oibuild/oi-userland/components/web/firefox/firefox-78.9.0/mozconfig ; echo "ac_add_options CXX=/usr/gcc/7/bin/g++" >> /oibuild/oi-userland/components/web/firefox/firefox-78.9.0/mozconfig ; echo "ac_add_options --prefix=/usr" >> /oibuild/oi-userland/components/web/firefox/firefox-78.9.0/mozconfig ; echo "ac_add_options --libdir=/usr/lib/amd64" >> /oibuild/oi-userland/components/web/firefox/firefox-78.9.0/mozconfig ; echo "ac_add_options --disable-av1" >> /oibuild/oi-userland/components/web/firefox/firefox-78.9.0/mozconfig ; echo "ac_add_options --disable-crashreporter" >> /oibuild/oi-userland/components/web/firefox/firefox-78.9.0/mozconfig ; echo "ac_add_options --disable-debug-symbols" >> /oibuild/oi-userland/components/web/firefox/firefox-78.9.0/mozconfig ; echo "ac_add_options --disable-install-strip" >> /dev/null ; echo "ac_add_options --disable-optimize" >> /dev/null ; echo "ac_add_options --disable-tests" >> /oibuild/oi-userland/components/web/firefox/firefox-78.9.0/mozconfig ; echo "ac_add_options --disable-updater" >> /oibuild/oi-userland/components/web/firefox/firefox-78.9.0/mozconfig ; echo "ac_add_options --enable-debug" >> /dev/null ; echo "ac_add_options --enable-js-shell" >> /oibuild/oi-userland/components/web/firefox/firefox-78.9.0/mozconfig ; echo "ac_add_options --enable-official-branding" >> /oibuild/oi-userland/components/web/firefox/firefox-78.9.0/mozconfig ; echo "ac_add_options --enable-optimize" >> /oibuild/oi-userland/components/web/firefox/firefox-78.9.0/mozconfig ; echo "ac_add_options --enable-pulseaudio" >> /oibuild/oi-userland/components/web/firefox/firefox-78.9.0/mozconfig ; echo "ac_add_options --enable-system-pixman" >> /oibuild/oi-userland/components/web/firefox/firefox-78.9.0/mozconfig ; echo "ac_add_options --enable-update-channel=esr" >> /oibuild/oi-userland/components/web/firefox/firefox-78.9.0/mozconfig ; echo "ac_add_options --with-intl-api" >> /oibuild/oi-userland/components/web/firefox/firefox-78.9.0/mozconfig ; echo "ac_add_options --with-system-libevent" >> /oibuild/oi-userland/components/web/firefox/firefox-78.9.0/mozconfig ; echo "ac_add_options --with-system-zlib" >> /oibuild/oi-userland/components/web/firefox/firefox-78.9.0/mozconfig ; echo "ac_add_options --host=x86_64-sun-solaris" >> /oibuild/oi-userland/components/web/firefox/firefox-78.9.0/mozconfig ; echo "ac_add_options --target=x86_64-sun-solaris" >> /oibuild/oi-userland/components/web/firefox/firefox-78.9.0/mozconfig ;  ( cd firefox-78.9.0; autoreconf; touch configure; cd js/src; autoreconf; touch configure; cargo install cbindgen );
    Updating crates.io index
 Downloading crates ...
  Downloaded cbindgen v0.19.0
  Installing cbindgen v0.19.0
error: failed to compile `cbindgen v0.19.0`, intermediate artifacts can be found at `/tmp/cargo-installEV17el`

Caused by:
  failed to parse lock file at: /oibuild/.cargo/registry/src/github.com-1ecc6299db9ec823/cbindgen-0.19.0/Cargo.lock

Caused by:
  invalid serialized PackageId for key `package.dependencies`
gmake: *** [/oibuild/oi-userland/make-rules/configure.mk:174: /oibuild/oi-userland/components/web/firefox/build/amd64/.configured] Error 101

**Comment by cgrzemba** [comment]
This is probably raised by an setup of old cargo in $HOME. It should be ok to remove the whole directory $HOME/.cargo and start the crate install again.

**Comment by AndWac** [comment]
Ok, that works (maybe we need to clean that when gmake clean is being run), but when I check the sample-manifest I get this differences:
╰─➤  diff sample-manifest.p5m sample-manifest.p5m.orig
26a27,28
> file path=usr/lib/$(MACH64)/firefox/browser/blocklist.xml
> file path=usr/lib/$(MACH64)/firefox/browser/chrome.manifest
33a36
> file path=usr/lib/$(MACH64)/firefox/browser/features/fxmonitor@mozilla.org.xpi
37a41
> file path=usr/lib/$(MACH64)/firefox/chrome.manifest
55a60,61
> file path=usr/lib/$(MACH64)/firefox/libnssdbm3.chk
> file path=usr/lib/$(MACH64)/firefox/libnssdbm3.so

Please update the sample-manifest.p5m file, too.

**Comment by AndWac** [comment]
gmake publish fails with
Couldn't find 'usr/lib/amd64/firefox/browser/blocklist.xml' in any of the specified search directories:
	/oibuild/oi-userland/components/web/firefox
	/oibuild/oi-userland/components/web/firefox/build
	/oibuild/oi-userland/components/web/firefox/build/prototype/i386
	/oibuild/oi-userland/components/web/firefox/build/prototype/i386/mangled
	/oibuild/oi-userland/components/web/firefox/files
	/oibuild/oi-userland/components/web/firefox/firefox-78.9.0
Couldn't find 'usr/lib/amd64/firefox/browser/chrome.manifest' in any of the specified search directories:
	/oibuild/oi-userland/components/web/firefox
	/oibuild/oi-userland/components/web/firefox/build
	/oibuild/oi-userland/components/web/firefox/build/prototype/i386
	/oibuild/oi-userland/components/web/firefox/build/prototype/i386/mangled
	/oibuild/oi-userland/components/web/firefox/files
	/oibuild/oi-userland/components/web/firefox/firefox-78.9.0
Couldn't find 'usr/lib/amd64/firefox/browser/features/fxmonitor@mozilla.org.xpi' in any of the specified search directories:
	/oibuild/oi-userland/components/web/firefox
	/oibuild/oi-userland/components/web/firefox/build
	/oibuild/oi-userland/components/web/firefox/build/prototype/i386
	/oibuild/oi-userland/components/web/firefox/build/prototype/i386/mangled
	/oibuild/oi-userland/components/web/firefox/files
	/oibuild/oi-userland/components/web/firefox/firefox-78.9.0
Couldn't find 'usr/lib/amd64/firefox/chrome.manifest' in any of the specified search directories:
	/oibuild/oi-userland/components/web/firefox
	/oibuild/oi-userland/components/web/firefox/build
	/oibuild/oi-userland/components/web/firefox/build/prototype/i386
	/oibuild/oi-userland/components/web/firefox/build/prototype/i386/mangled
	/oibuild/oi-userland/components/web/firefox/files
	/oibuild/oi-userland/components/web/firefox/firefox-78.9.0
Couldn't find 'usr/lib/amd64/firefox/libnssdbm3.chk' in any of the specified search directories:
	/oibuild/oi-userland/components/web/firefox
	/oibuild/oi-userland/components/web/firefox/build
	/oibuild/oi-userland/components/web/firefox/build/prototype/i386
	/oibuild/oi-userland/components/web/firefox/build/prototype/i386/mangled
	/oibuild/oi-userland/components/web/firefox/files
	/oibuild/oi-userland/components/web/firefox/firefox-78.9.0
Couldn't find 'usr/lib/amd64/firefox/libnssdbm3.so' in any of the specified search directories:
	/oibuild/oi-userland/components/web/firefox
	/oibuild/oi-userland/components/web/firefox/build
	/oibuild/oi-userland/components/web/firefox/build/prototype/i386
	/oibuild/oi-userland/components/web/firefox/build/prototype/i386/mangled
	/oibuild/oi-userland/components/web/firefox/files
	/oibuild/oi-userland/components/web/firefox/firefox-78.9.0
gmake: *** [/oibuild/oi-userland/make-rules/ips.mk:472: /oibuild/oi-userland/components/web/firefox/build/manifest-i386-firefox.depend] Error 1

So, the firefox.p5m file also needs adoptions.

**Comment by Toasterson** [comment]
> This is probably raised by an setup of old cargo in $HOME. It should be ok to remove the whole directory $HOME/.cargo and start the crate install again.

You can also set --root and --target-dir to any directory you want to override the installtion into $HOME. Might be usefull to keep cbindgen close to the firefox installation as we do not package it.

**Comment by pyhalov** [comment]
If it's not rquired in runtime, install it somewhere under BUILD_DIR.

**Comment by cgrzemba** [comment]
I fixed the language files location too.

File: components/web/firefox/Makefile [hunk 1]
```diff
@@ -10,96 +10,150 @@
...
-include $(WS_MAKE_RULES)/ips.mk
+CONFIGURE_DEFAULT_DIRS=no
+
+include $(WS_MAKE_RULES)/common.mk
 
-PATH=$(USRBINDIR.64):$(PATH.illumos)
+#PATH=$(USRBINDIR.64):$(PATH.illumos)
+# /usr/gnu/bin needed to pick up the GNU version of tail and readelf.
+# /usr/sbin needed for dtrace.
+PATH = /usr/gnu/bin:/usr/bin:/usr/sbin
```
Comment (AndWac) [inline components/web/firefox/Makefile]: We have PATH.gnu (vs. PATH.illumos) for this so
PATH = $(PATH.gnu)
should be used.

File: components/web/firefox/Makefile [hunk 2]
```diff
@@ -10,96 +10,150 @@
...
 
-COMPONENT_PREP_ACTION = ( cd $(@D) && autoconf-2.13 &&\
-                          cd $(@D)/js/src && autoconf-2.13 )
+GCC_VERSION= 7
```
Comment (AndWac) [inline components/web/firefox/Makefile]: gcc-7 is our momentary default. That will probably change in the future and then gcc-10 might be the default...

**Review by AndWac — APPROVED** [review]
I have successfully built and installed it.
Now I will test its memory consumption et al.
When I am happy with it I will merge the PR.

File: components/web/firefox/patches/Bug577056-1.patc [hunk 3]
```diff
@@ -0,0 +1,49 @@
+diff --git a/js/public/Value.h b/js/public/Value.h
```
Comment (pyhalov) [inline components/web/firefox/patches/Bug577056-1.patc:1]: This file (components/web/firefox/patches/Bug577056-1.patc) is unused

File: components/web/firefox/patches/Bug577056.patc [hunk 4]
```diff
@@ -0,0 +1,50 @@
+diff --git a/components/web/firefox/firefox-68.12.0/js/mapfile.IllumOS b/js/mapfile.IllumOS
```
Comment (pyhalov) [inline components/web/firefox/patches/Bug577056.patc:1]: This file is also unused - is it intentional?

Comment (cgrzemba) [reply components/web/firefox/patches/Bug577056.patc:1]: yes I will remove these two files. This was the attempt to get the tagged pointer things running.

**Pull request merged**